src/app-layout/utils/use-resize.tsx

```diff
@@ -0,0 +1,134 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useState, useEffect, useMemo, useCallback } from 'react';
+
+import { fireNonCancelableEvent } from '../../internal/events';
+import clsx from 'clsx';
+
+import ResizeHandler from '../../split-panel/icons/resize-handler';
+import { getLimitedValue } from '../../split-panel/utils/size-utils';
+import { usePointerEvents } from './use-pointer-events';
+import { useKeyboardEvents } from './use-keyboard-events';
+import { DrawersProps } from '../visual-refresh/drawers';
+
+import splitPanelStyles from '../../split-panel/styles.css.js';
+import testutilStyles from '../test-classes/styles.css.js';
+import styles from '../visual-refresh/styles.css.js';
+import { DrawerFocusControlRefs } from './use-drawer-focus-control';
+
+export interface SizeControlProps {
+  position: 'side';
+  panelRef?: React.RefObject<HTMLDivElement>;
+  handleRef?: React.RefObject<HTMLDivElement>;
+  setSidePanelWidth: (width: number) => void;
+  setBottomPanelHeight: (height: number) => void;
+}
+
+export interface DrawerResizeProps {
+  activeDrawerId: string | null;
+  drawers: DrawersProps;
+  drawersRefs: DrawerFocusControlRefs;
+  isToolsOpen: boolean;
+  drawersMaxWidth: number;
+}
+
+function useResize(drawerRefObject: React.RefObject<HTMLDivElement>, drawerResizeProps: DrawerResizeProps) {
+  const { activeDrawerId, drawers, drawersRefs, isToolsOpen, drawersMaxWidth } = drawerResizeProps;
+
+  const activeDrawer = drawers?.items.find((item: any) => item.id === activeDrawerId) ?? null;
```

Comment:
Can we remove `any` here, to increase the type safety? (see also other comment below)